engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -485,6 +485,10 @@ protected void deactivateSinks()
 
   void checkpoint(long windowId)
   {
+    if (operator instanceof Operator.CheckpointNotificationListener) {
+      ((Operator.CheckpointNotificationListener)operator).beforeCheckpoint(windowId);
```

Comment:
The checkpoint call is called from within the Node lifecycle run method which has a global handling for Throwable.